### PR TITLE
Set up libgc for large config

### PR DESCRIPTION
**Pull request by DanGrayson**
see issue #2445

### CODE REVIEW

**Comment by mikestillman** [comment]
@DanGrayson I can't tell what files were actually part of this pull request? (Most of the files seem to be from other things?)  In particular, how did you fix it?

**Comment by DanGrayson** [comment]
> @DanGrayson I can't tell what files were actually part of this pull request? (Most of the files seem to be from other things?) In particular, how did you fix it?

Oops, the base was "master" instead of "development".  Now you can tell.

**Comment by mikestillman** [comment]
@DanGrayson @d-torrance Is it ok if we by default build gc?  Doug, how would you handle that with debian/ubuntu?
@kschwede I would love to see if this works for 8 TB of RAM!  I can give you a problem that will quickly use a large amount of ram (although, you might have such examples too), but I don't know if these examples will use that much ram or not...! (But probably at least one will).

**Comment by mikestillman** [comment]
@d-torrance Maybe packaging systems should start using the large config, as there are many machines with even minimally large memory available (stopping/crashing after 8 GB of RAM seems extremely small these days).

**Comment by d-torrance** [comment]
In Debian/Ubuntu, we need to use the system gc package, so I'll either patch the Macaulay2 package to use it or, if a configure option is available instead, use that.

I'm not the gc maintainer in Debian, but I can file a wishlist bug to bump up the memory allowance.

**Comment by kschwede** [comment]
> @kschwede I would love to see if this works for 8 TB of RAM! I can give you a problem that will quickly use a large amount of ram (although, you might have such examples too), but I don't know if these examples will use that much ram or not...! (But probably at least one will).

Sure, I can probably try this on one of the machines with 6 TB here?  If you have an example you'd like to try, please go ahead and send it to me.

File: M2/libraries/gc/Makefile.in [hunk 1]
```diff
@@ -75,7 +78,7 @@ CONFIGOPTIONS +=  --enable-parallel-mark
 # INSTALLTARGET = install-strip
 endif
 
-CONFIGOPTIONS += --enable-large-config --disable-gcj-support --disable-java-finalization
+CONFIGOPTIONS += --disable-gcj-support --disable-java-finalization
```
Comment (jkyang92) [inline M2/libraries/gc/Makefile.in:81]: Why not just use --enable-large-config. It's going to set MAX_HEAP_SECTS to 81920 anyways (at least on 64bit systems, 32bit programs can't access that much memory). Also it sets several other values to larger values relevant for larger memory usage.

Comment (DanGrayson) [reply M2/libraries/gc/Makefile.in:81]: Hmm, maybe.  But could it really make a difference?

**Pull request merged**